### PR TITLE
Update to use non-deprecated convex types

### PR DESCRIPTION
**Pull request by sshader**
Some of the types we were using were marked deprecated in 1.1.1, and removed fully in 1.11.0.

This PR changes us to use the non-deprecated versions and require a peer dependency of 1.1.1 or higher, which I think means this should work with 1.1.1 and upwards.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "lib": ["ESNext", "DOM", "DOM.Iterable"],
     "module": "ESNext",
     "skipLibCheck": true,
+    "noEmit": true,
```
Comment (sshader) [inline tsconfig.json:8]: Setting this because I wanted to `npx tsc -p .` to check that everything typechecked and did not want to emit the JS files

**Review by sshader — COMMENTED** [review]
Assuming we want this, do I just `npm publish` from `main` once it's landed?

File: tsconfig.json (hunk 1 above)
Comment (xixixao) [reply tsconfig.json:8]: Just make sure the build command still works.

**Review by xixixao — APPROVED** [review]
I need to kill this lib.

File: tsconfig.json (hunk 1 above)
Comment (sshader) [reply tsconfig.json:8]: I ran it which produced different `react.js.map` files which are fairly inscrutable. Lmk if there's something else I should check.

**Pull request merged**